DataFormats/SiPixelDigiSoA/interface/alpaka/SiPixelDigisSoACollection.h

```diff
@@ -23,7 +23,7 @@ namespace cms::alpakatools {
   struct CopyToHost<SiPixelDigisDevice<TDevice>> {
     template <typename TQueue>
     static auto copyAsync(TQueue &queue, SiPixelDigisDevice<TDevice> const &srcData) {
-      SiPixelDigisHost dstData(srcData.view().metadata().size(), queue);
+      SiPixelDigisHost dstData(srcData.nDigis(), queue);
```

Comment:
Isn't this fix actually critical for correctness? I mean, if the `nDigis` happens to be on the edge of the padding (i.e. `nDigis % 128 == 0` and thus `nDigis+1` would lead to a larger padded memory allocation than `nDigis`), the `View`s of `srcData` and `dstData` place the column pointers to different places in the buffer.

In other words, with 1D `memcpy()` it is imperative to have source and destination PortableCollections to have the same size.

Reply:
Anyway this fix was indeed "critical" for Phase2 wfs at least. I saw random events failing due to the meaningless RawId in the digis SoA independently of the event size (SingleMu noPU or TTbar PU). What's odd is that, no matter how many events I have run on, I've never seen a problem on Run3.

Reply:
(and  all the events crashing had `(nDigis+1) % 64 == 0` actually)

Reply:
In Run 3, `nDigis` is guaranteed to be even¹, so it can never happen that `(nDigis + 1) % 64 == 0`.

---
¹ see `wordCounter` in https://github.com/cms-sw/cmssw/blob/b3830cced4787ed0573f42a7fa42644607013803/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc#L231-L241

Reply:
Ah! Never noticed that, thanks. What a coincidence (for the point of view of this very specific bug).